pkg/database/mobile_app.go

```diff
@@ -133,20 +133,30 @@ func (a *MobileApp) BeforeSave(tx *gorm.DB) error {
 	return nil
 }
 
-// ListActiveApps finds all active mobile apps.
-func (db *Database) ListActiveApps() ([]*MobileApp, error) {
-	// Find the apps.
-	var apps []*MobileApp
-	if err := db.db.
-		Model(&MobileApp{}).
-		Find(&apps).
-		Order("mobile_apps.deleted_at DESC, LOWER(mobile_apps.name)").
-		Error; err != nil {
-		if IsNotFound(err) {
-			return apps, nil
-		}
+// ExtendedMobileApp combines a MobileApp with its Realm
+type ExtendedMobileApp struct {
+	MobileApp
+	Realm
+}
+
+// ListActiveAppsWithRealm finds all active mobile apps with their associated realm.
+func (db *Database) ListActiveAppsWithRealm() ([]*ExtendedMobileApp, error) {
+	rows, err := db.db.Table("mobile_apps").
+		Select("mobile_apps.*, realms.*").
+		Joins("left join realms on realms.id = mobile_apps.realm_id").
```

Comment:
There's a subtle thing here... if a realm is deleted, but the associated mobile app is not, you'd have a NPE later.

Reply:
I don't believe so. We always make an instance of ExtendedMobileApp and realm is inlined